sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -226,6 +234,11 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
       }
      """
   }
+
+  override def sql: Option[String] = for {
+    valueSQL <- child.sql
+    listSQL <- sequenceOption(hset.toSeq.map(Literal(_).sql))
+  } yield s"($valueSQL IN (${listSQL.mkString(", ")}))"
```

Comment:
yeah. But, does `Literal(_)` work if hset's elements are in internal types (e.g. UTF8String)?